src/PostgREST/Config.hs

```diff
@@ -196,6 +198,14 @@ readPathShowHelp = customExecParser parserPrefs opts
           |
           |## logging level, the admitted values are: crit, error, warn and info.
           |# log-level = "error"
+          |
+          |## rollback all transactions by default, use for test environments
+          |## disabled by default
+          |# transaction-rollback = false
```

Comment:
How about having this one only for internal use?

It would only be declared on the `Config` type. As an example, we used to have this setting:

https://github.com/PostgREST/postgrest/blob/96a16a377f8dab58a72cb5c6d7ddffee2316334e/src/PostgREST/Config.hs#L92

That one disabled logging and it was only set to True on the tests.

Reply:
Though I have a question here. Would the rollback only make sense in a global way? Or do we need it to be on demand?

Perhaps we just need one of these options.

Reply:
> How about having this one only for internal use?

But.. but... but... I want to use this in my project's dev environment as well. I run automated tests there. Some via `pgtap` directly in the database, but also some that make requests via `newman` through PostgREST. Those should not modify the database by default. I don't want to put `tx=rollback` on *every* request.

> Though I have a question here. Would the rollback only make sense in a global way? Or do we need it to be on demand?
> 
> Perhaps we just need one of these options.

But then I have some requests, that I need to commit. So I need default=rollback / override=commit for sure.

And I can very well imagine both the following scenarios as well:
- default=rollback / no overrides: Public PostgREST demo server to test requests against. Maybe even from a webinterface, linked to in the docs. Aka "read the docs, try it out immediately". That would be really cool.

- default=commit / override=rollback: Maybe I do want to make a request on a production server for diagnostic purposes. It must be rolled back, but all the other requests that other users make, should go through. I can even imagine using that somehow to validate requests before they are actually made. Think of a form on the client, where the user can enter stuff. You could even "test" the request in the background, before they submit, and use error messages to display validation hints. Only if the user hits submit, you submit the real request, though.

I am sure people will find more use-cases for this! :)

Reply:
Oh, cool cool :). The demo server idea sounds amazing!

Then I think what we should do is to group the options, by prefix. How about naming them like:

- rollback-all: all requests will be rolled back
- rollback-on-prefer: rollback on demand with Prefer

Reply:
I like grouping them, but I don't think `rollback-` is correct. Because on `rollback-all = true`, the `prefer` setting actually allows `prefer tx=commit`, not rollback.

Maybe:
- `prefer-tx-allowed` and
- `prefer-tx-default`

But this is not a 100%. This could be interpreted as `default` only kicking in when `allowed: true` or something like that.

What about just one config option `transactions = force-commit | commit | rollback | force-rollback`, where `force-commit` is the default? Better name for "transactions" maybe?

Reply:
`tx-prefer-allowed` and `tx-rollback-by-default` could also work.

Reply:
>  Because on rollback-all = true, the prefer setting actually allows prefer tx=commit, not rollback.

I see. Nice, that's a sensible behavior.

> tx-prefer-allowed and tx-rollback-all

I prefer this one. It's more obvious. The `transactions` option is not so straightforward.

`tx-prefer-allowed` seems related to all prefers though, but I don't have a better name for now. It should be enough to clarify it with the option description.

(Or maybe `tx-override-on-prefer`?)

Reply:
I went with `tx-rollback-all` and `tx-allow-override` and improved the help output about it a bit.